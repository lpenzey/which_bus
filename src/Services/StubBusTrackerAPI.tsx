import sampleResponse from './__fixtures__/searchResponse.json';

const StubBusTrackerAPI = {
    requestTimeEstimate: async (route: number, stpid: number, format: string) => {
        return sampleResponse;
    }
};

export default StubBusTrackerAPI;