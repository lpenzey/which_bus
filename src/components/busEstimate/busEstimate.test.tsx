import React from 'react';
import BusEstimate from './busEstimate';
import StubBusTrackerAPI from '../../Services/stubBusTrackerAPI';
import { render, fireEvent, cleanup, waitForElement } from '@testing-library/react'

afterEach(cleanup)
it("inserts estimate time", async () => {
    const { getByText, getByTestId } = render(<BusEstimate api={StubBusTrackerAPI} />);
    fireEvent.click(getByText('Get Estimate'))

    const greetingTextNode = await waitForElement(() =>
        getByTestId('estimate')
    )
    expect(greetingTextNode).toMatchSnapshot()
});

