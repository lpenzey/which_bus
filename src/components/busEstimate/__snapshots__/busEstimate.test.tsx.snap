// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inserts estimate time 1`] = `
<div
  class="innerText"
  data-testid="estimate"
>
  ROUTE: 20
  <br />
  STOP: Madison & Jefferson
  <br />
  DIRECTION: Westbound
  <br />
  ARRIVING AT: 12:02
  <br />
</div>
`;
